posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -0,0 +1,343 @@
+----
+
+* The `enableImplicitBeanArchives` attribute is the same as it was in previous versions.
+  - If set to `true`, which is the default, then archives that do not contain a `beans.xml` file are treated as Implicit Bean Archives and scanned for classes that have Bean Defining Annotations.
+  - If set to `false`, then archives that do not contain a `beans.xml` file are not scanned for annotated classes.
+
+* The `emptyBeansXmlCDI3Compatibility` attribute applies only to CDI 4.0.
+  - If set to `true`, then an archive that contains an empty `beans.xml` file is treated as an explicit bean archive, as it was in CDI 3.0 and earlier.
+  - If set to `false`, which is the default, then an archive that contains an empty `beans.xml` file is treated as an implicit bean archive. This is equivalent to setting `bean-discovery-mode="annotated"`..
+
+To use the CDI 4.0 feature, add `cdi-4.0` to your `server.xml`:
+
+[source, xml]
+----
+<featureManager>
+  <feature>cdi-4.0</feature>
+</featureManager>
+----
+
+[#expression]
+== Jakarta EE 10 Expression Language 5.0
+
+Jakarta EE 10 Expression Language 5.0 includes a number of new features, deprecations, and specification clarifications.  One of the new features is the addition of the link:https://jakarta.ee/specifications/expression-language/5.0/apidocs/jakarta.el/jakarta/el/methodreference[MethodReference] which "Provides information about the method to which a method expression resolves." This new class allows developers to access any annotations present on a particular method using the `getAnnotations()` method.
+
+To enable Expression Language 5.0, add the `expressionLanguage-5.0` feature to the list of features in your server.xml file:
+[source,xml]
+----
+<featureManager>
+  <feature>expressionLanguage-5.0</feature>
+</featureManager>
+----
+
+More information about Expression Language 5.0 can be found in: 
+
+* link:https://jakarta.ee/specifications/expression-language/5.0/jakarta-expression-language-spec-5.0.html#changes-between-5-0-and-4-0[Changes between the Expression Language 4.0 and Expression Language 5.0]
+
+* link:https://jakarta.ee/specifications/expression-language/5.0/[Expression Language 5.0 specification]
+
+* link:{url-prefix}/docs/latest/reference/feature/expressionLanguage.html[Open Liberty Expression Language feature documentation]
+
+[#microprofile]
+== MicroProfile OpenAPI 3.1 
+
+The link:https://swagger.io/specification/[OpenAPI specification] standardizes a way of documenting REST APIs in a JSON or YAML format. The link:https://download.eclipse.org/microprofile/microprofile-open-api-3.1-RC2/microprofile-openapi-spec-3.1-RC2.html[MicroProfile OpenAPI specification] helps you generate and serve OpenAPI documentation for your REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people using the API in production.
```

Comment:
```suggestion
The link:https://swagger.io/specification/[OpenAPI specification] standardizes a way of documenting REST APIs in a JSON or YAML format. The link:https://download.eclipse.org/microprofile/microprofile-open-api-3.1-RC2/microprofile-openapi-spec-3.1-RC2.html[MicroProfile OpenAPI specification] helps you generate and serve OpenAPI documentation for your REST applications that are built by using Jakarta RESTful Web Services. This can be useful for developers to test out the API during development, or for people using the API in production.
```

Reply:
The reason I did that is because mpOpenAPI 3.1 does not work with Java / Jakarta EE 8 where it is called JAX-RS.  It only works with EE9 only today but eventually EE 10 only.

Reply:
With that in mind do we need a statement about mpOpenAPI-3.1 only working with Jakarta EE 9 features and that it current doesn't work with Jakarta EE 10 because not all EE 10 features needed for mpOpenAPI feature are available yet in the beta?

Reply:
I guess it depends how one reads the sentence; "MicroProfile OpenAPI" broadly (in which case both apply) or specific to 3.1 (in which case just Jakarta would apply).
I'm wondering if I should leave it as is, and just point to a version agnostic link like
```suggestion
The link:https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md[OpenAPI specification] standardizes a way of documenting REST APIs in a JSON or YAML format. The link:https://microprofile.io/project/eclipse/microprofile-open-api[MicroProfile OpenAPI specification] helps you generate and serve OpenAPI documentation for your REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people using the API in production.
```

Reply:
Can possibly do that.  but it should be Jakarta RESTful Web Services not Jakarta restfulWS

Reply:
udpated